# This file sets the values for the static parameters along with the some of 
# the default values for the dynamic parameters


# Setting the defaults for the dynamic parameters server
# NOTE: this overrides the defaults set in the Parameters.cfg file
dynamic_parameters_server:
  re_algorithm: "melosee"  # Retinal Encoder (re)
  sg_algorithm: "melosee"  # Sound Generator (sg)
  # Below specify general dynamic parameters used for algorithms
  num_quantization_levels: 8


depth_camera:
  min_depth: # in meters
  max_depth: # in meters
  h_fov: -1
  v_fov: -1
  # For cropping, set the % of dead zone to remove (set to 0 for no cropping)
  # e.g. a value of 0.2 would crop 10% off the left and 10% off the right
  crop_width: 0.1 
  crop_height: 0


color_camera:
  h_fov: -1
  v_fov: -1
  # For cropping, set the % of dead zone to remove (set to 0 for no cropping)
  # e.g. a value of 0.2 would crop 10% off the left and 10% off the right
  crop_width: 0  
  crop_height: 0


# Preprocessor static values
pp:
  input:
    depth_image:
      topic: "camera/depth/image_rect_raw"
  output:
    depth_image:
      topic: "processed_depth_image"
      width: 96 # pixels


# Retinal Encoder static values
re:
  input:
    depth_image:
      topic: "processed_depth_image"
  output:
    depth_image:
      topic: "retinal_encoded_image"
      width: 9 # pixels


# Sound Generator static values
sg:
  input:
    retinal_encoded_image:
      topic: "retinal_encoded_image"